apps/XMRig/install

```diff
@@ -0,0 +1,287 @@
+      tls="$(sed -n 4p <<<"$output" | sed 's/Disabled/false/' | sed 's/Supported by mining pool/true/')"
+      keepalive="$(sed -n 5p <<<"$output" | sed 's/Disabled/false/' | sed 's/Supported by mining pool/true/')"
+      
+      if [ -z "$wallet_address" ];then
+        warning "Wallet address cannot be left blank!"
+      elif [ -z "$mining_pool" ];then
+        warning "Mining pool cannot be left blank!"
+      else
+        local save_settings='yes'
+        break
+      fi
+    else
+      local save_settings='no'
+      break
+    fi
+  done
+  
+  if [ "$save_settings" == 'yes' ];then
+    # Save choices to xmrig.json
+    status "Saving to $HOME/.config/xmrig.json"
+    echo "Contents of file:"
+    cat <<EOT | tee $HOME/.config/xmrig.json 1>&2 || error 'Failed to create $HOME/.config/xmrig.json!'
+{
+    "autosave": true,
+    "background": null,
+    "cpu": true,
+    "opencl": false,
+    "cuda": false,
+    "rx": [-1, -1],
+    "pools": [
+        {
+            "url": "${mining_pool}",
+            "user": "${wallet_address}",
+            "rig-id": "${miner_name}",
+            "keepalive": ${keepalive},
+            "tls": ${tls}
+        }
+    ]
+}
+EOT
+  else #User skipped setup wizard
+    echo -e "You chose to not set up a configuration for xmrig."
+  fi
+  
+} # End setup_wizard()
+
+cd
+rm -rf $HOME/xmrig
+
+#Compile xmrig; set to false to skip compilation
+if true;then
+  install_packages git build-essential cmake libuv1-dev libssl-dev libhwloc-dev yad xprintidle || exit 1
+  
+  git_clone https://github.com/MoneroOcean/xmrig || error 'Failed to clone repository!'
+  
+  mkdir $HOME/xmrig/build; cd $HOME/xmrig/build || error 'Failed to create and cd into build directory!'
+  
+  status "Preparing to compile..."
+  
+  #Optimize for specific CPU architecture, if detected. Get full list with 'gcc --target-help -mtune=foo' 
+  if lscpu | grep -q Cortex-A72 ;then
+    #Host device is using cortex-a72 CPU - this includes Pi4
+    mtune='-mtune=cortex-a72'
+  elif lscpu | grep -q Cortex-A53 ;then
+    #Host device is using cortex-a53 CPU - this includes Pi3
+    mtune='-mtune=cortex-a53'
+  fi
+  
+  if [ "$arch" == 32 ];then
+    DARM_TARGET=7
+  elif [ "$arch" == 64 ];then
+    DARM_TARGET=8
+  else
+    error "Failed to detect OS CPU architecture! Something is very wrong."
+  fi
+  
+  #Cmake flags:
+  # -O3: enables many performance optimizations at the expense of compilation time. From: https://www.youtube.com/watch?v=4e0fRKHG7Hk
+  # -mtune: tunes the program for current system's detected CPU. From: https://github.com/xmrig/xmrig/issues/446#issuecomment-373450600
+  # -ffast-math: speeds up math and seems to be stable for xmrig.
+  # -funsafe-loop-optimizations: Assume no overflowing loops
+  # -DARM_TARGET: selects ARM processor major architecture - either 7 or 8
+  
+  cmake .. -O3 -DCMAKE_CXX_FLAGS="$mtune -ffast-math -funsafe-loop-optimizations" -DARM_TARGET="$DARM_TARGET" || error 'Failed to generate build files!'
```

Comment:
xmrig contains C and C++ code, you need to add the args to both `-DCMAKE_C_FLAGS` and `-DCMAKE_CXX_FLAGS`

Reply:
Done.